mongo/options/indexoptions.go

```diff
@@ -116,162 +122,208 @@ func MergeListIndexesOptions(opts ...*ListIndexesOptions) *ListIndexesOptions {
-	Background              *bool
-	ExpireAfterSeconds      *int32
-	Name                    *string
-	Sparse                  *bool
-	StorageEngine           interface{}
-	Unique                  *bool
-	Version                 *int32
-	DefaultLanguage         *string
-	LanguageOverride        *string
-	TextVersion             *int32
-	Weights                 interface{}
-	SphereVersion           *int32
-	Bits                    *int32
-	Max                     *float64
-	Min                     *float64
-	BucketSize              *int32
+	// If true, the index will be built in the background on the server and will not block other tasks. The default
+	// value is false.
+	Background *bool
+
+	// The length of time, in seconds, for documents to remain in the collection. The default value is 0, which means
+	// that documents will remain in the collection until they're explicitly deleted or the collection is dropped.
+	ExpireAfterSeconds *int32
+
+	// The name of the index. The default value is "[field1]_[direction1]_[field2]_[direction2]...". For example, an
+	// index with the specification {name: 1, age: -1} will be named "name_1_age_-1".
+	Name *string
+
+	// If true, the index will only reference documents that contain the fields specified in the index. The default is
+	// false.
+	Sparse *bool
+
+	// Specifies the storage engine to use for the index. The value must be a document in the form
+	// {<storage engine name>: <options>}. The default value is nil, which means that the default storage engine
+	// will be used. This option is only applicable for MongoDB versions >= 3.0 and is ignored for previous server
+	// versions.
+	StorageEngine interface{}
+
+	// If true, the collection will not accept insertion or update of documents where the index key value matches an
+	// existing value in the index. The default is false.
+	Unique *bool
+
+	// TODO: not sure what this does
+	Version *int32
+
+	// The language that determines the list of stop words and the rules for the stemmer and tokenizer. This option
+	// is only applicable for text indexes and is ignored for other index types. The default value is "english".
+	DefaultLanguage *string
+
+	// The name of the field in the collection's documents that contains the override language for the document. This
+	// option is only applicable for text indexes and is ignored for other index types. The default value is the value
+	// of the DefaultLanguage option.
+	LanguageOverride *string
+
+	// TODO: not sure what this does.
+	TextVersion *int32
+
+	// A document that contains field and weight pairs. The weight is an integer ranging from 1 to 99,999, inclusive,
+	// indicating the significance of the field relative to the other indexed fields in terms of the score. This option
+	// is only applicable for text indexes and is ignored for other index types. The default value is nil, which means
+	// that every field will have a weight of 1.
+	Weights interface{}
+
+	// TODO: not sure what this does.
```

Comment:
See https://docs.mongodb.com/manual/core/2dsphere/#dsphere-v2

Reply:
Added a doc comment with a link to that site.